### DISCUSSION TITLE
No Connection with WIFI ELM327 Adapter from Kingbolen

### DISCUSSION THREAD

**Discussion by videejay** [cedricp/ddt4all#1101] [Q&A]
Hi everyone,

I purchased an ELM327 WIFI Dongle with pic18f25k80 from Kingbolen (https://kingbolentool.com/products/kingbolen-obd2-scanner-bluetooth-check-engine-light) which I now tried to connect with ddt4all but unfortunately without success. With iOS EOBD it is working fine. Also with Windows App like PCMSCAN it can connect to my car (Renault Megane 3).

When opening ddt4all and clicking on Test connection it takes an eternity trying to connect to 192.168.0.10:35000 at baudrate 38k (WiFi connected to the Dongle obviously)
I tried using 32bit and 64Bit Version (latest release) on a Win10 laptop. 

Any suggestions on how to debug this? 

Thanks for your support

**Comment by videejay** [comment]
Little update from my side.

I managed to connect using ScanMaster ELM or PCMScan in parallel to DDT4ALL.

1. Connect to WIFI of Dongle
1. Starting to connect via DDT4ALL (trying to connect to 192.168.0.10:35000 with 38k Baud) 
2. Start ScanMaster ELM Demo and connect (you need to first set Settings to WIFI for the Dongle)
3. After connecting immediatly close ScanMaster ELM

=> Connection in DDT4ALL works.

Seems like this is an init issue not sending the right commands to WIFI connected ELM Devices to me but not shure.

After that I can jump through all available ECUs and read out data (did not try to write yet)

![image](https://github.com/cedricp/ddt4all/assets/34542108/f4b80dc9-418d-4a0a-9210-d9168e1a6f79)
![image](https://github.com/cedricp/ddt4all/assets/34542108/2718fd54-156f-4e55-9a90-7cb4ab69e8d5)
![image](https://github.com/cedricp/ddt4all/assets/34542108/8ba01103-167e-4089-a880-3345dcd27cf0)
![image](https://github.com/cedricp/ddt4all/assets/34542108/b238960c-a982-4268-ad3a-8598396f6dac)

Any way on to find out what commands are missing which ScanMaster or PCMSCAN are sending in the beginning to initiate the ECU communicatino with the device?

Kind regards,
videejay

**Comment by videejay** [comment]
Writing was also successful. I could enable Andoid Auto and Rear Camera View with MFD v5.1 on my Megane 3 Phase 2

**Comment by pedja76** [comment]
Thank you.
Your method of connecting with the ScanMaster ELM Demo has helped me connect and write successfully. Reno Grand Scenic 3. Now you can start exploring the DDT4All application.

Best regards.
Pedja.

**Reply by wonka929** [reply]
My Wifi OBD interface accepts just 1 wifi connection so i could not do the same test.
I tried opening an OBD diagnostic software on the PC, and immediatly start DDT4ALL but it didn't work.
I'm out of ideas.

**Reply by pedja76** [reply]
Have you tried following this sequence of actions?
First, establish communication of your OBD interface with ScanMaster ELM Demo, wait for ScanMaster ELM Demo to load the necessary drivers, then launch the DDT4ALL tool. 
This is how I successfully managed to run the DDT4ALL tool on my laptop, both then and every time afterwards. 
Provided that the OBD interface has the appropriate PIC18f25k80 microcontroller.
